egs2/chime7/task1/local/data.sh

```diff
@@ -9,29 +10,245 @@ log() {
+        exit 1;
+    fi
+
+    if [ -d "${chime6_corpus}" ]; then
+        echo "${chime6_corpus} already exists,
+        exiting as I am assuming it has already been created."
+    else
+        log "Generate CHiME6 corpus from CHiME5"
+        local/generate_chime6_data.sh \
+            --cmd "${train_cmd}" \
+            ${CHIME5} \
+            ${chime6_corpus}
+    fi
+fi
+
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ] && ! contains ${skip_stages} 2; then
+    log "Stage 2: generate CHiME7 dataset"
+    if [ -d "${chime7_corpus}" ]; then
+        echo "${chime7_corpus} already exists"
+        exit
+    fi
+
+    python local/generate_chime7_data.py \
+        -c ${chime6_corpus} \
+        -d ${DIPCO} \
+        -m ${MIXER6} \
+        -o ${chime7_corpus}
+fi
+
+
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ] && ! contains ${skip_stages} 3; then
+    log "Stage 3: Parse data to Lhotse manifests"
+    for dset in chime6 dipco mixer6; do
+        for part in train dev; do
+        if [ ${dset} == dipco ] && [ ${part} == train ]; then
+            continue # dipco has no train set
+        fi
+        echo "Creating lhotse manifests for ${dset} in ${manifests_root}/${dset}"
+        python local/get_lhotse_manifests.py -c ${chime7_corpus} \
+            -d ${dset} \
+            -p ${part} \
+            -o ${manifests_root} \
+            --ignore_shorter 0.2
+        done
+    done
+fi
+
+
+if [ ${stage} -le 4 ] && [ ${stop_stage} -ge 4 ] && ! contains ${skip_stages} 4; then
+    log "Stage 4: Running GSS"
+    # check if GSS is installed, if not stop, user must manually install it
+    if [ ! command -v gss &> /dev/null ]; then
+        echo "GPU-based Guided Source Separation (GSS) could not be found,
+        please refer to the README for how to install it. \n
+        See also https://github.com/desh2608/gss for more informations."
+        exit
+    fi
+
+    for dset in chime6 dipco mixer6; do
+        for part in train dev; do
+            if [ ${dset} == dipco ] && [ ${part} == train ]; then
+                continue # dipco has no train set
+            fi
+
+            echo "Running Guided Source Separation for ${dset}/${part}, results will be in ${gss_dump_root}/${dset}/${part}"
+
+            local/run_gss.sh \
+                --manifests-dir ${manifests_root} \
+                --dset-name ${dset} \
+                --dset-part ${part} \
+                --exp_dir ${gss_dump_root} \
+                --cmd ${cuda_cmd} \
+                --nj ${ngpu} \
+                --max_batch_dur ${max_batch_dur}
+
+            echo "Guided Source Separation processing for ${dset}/${part} was successful !"
+            echo "Parsing the GSS output to lhotse manifests which will be placed in ${manifests_root}/${dset}/${part}"
+        done
+    done
+fi
+
+
+if [ ${stage} -le 5 ] && [ $stop_stage -ge 5 ]; then
+    # Preparing ASR training and validation data;
+    for dset in chime6 dipco; do
```

Comment:
This part is from sub_track1.sh stage 3. The original for loops didn't close. I don't know if the logic here is correct.